weave-js/src/core/ops/primitives/date.ts

```diff
@@ -36,6 +36,87 @@ export const opDateSub = makeDateOp({
   },
 });
 
+const momentDurations: moment.unitOfTime.DurationConstructor[] = [
+  'years',
+  'months',
+  'days',
+  'hours',
+  'minutes',
+  'seconds',
+  'milliseconds',
+];
+
+export const opTimestampDiffStringFormat = makeDateOp({
+  hidden: true,
+  name: 'date-diffDaysStringFormat',
+  argTypes: {lhs: {type: 'timestamp'}, rhs: {type: 'timestamp'}},
+  description: `Returns the difference between two millisecond ${docType(
+    'timestamp',
+    {
+      plural: true,
+    }
+  )} in days in a nice string form`,
+  argDescriptions: {
+    lhs: `First ${docType('timestamp')}`,
+    rhs: `Second ${docType('timestamp')}`,
+  },
+  returnValueDescription: `The difference between the two ${docType(
+    'timestamp',
+    {
+      plural: true,
+    }
+  )} in days in a nice string form`,
+  returnType: inputTypes => 'string',
+  resolver: inputs => {
+    if (inputs.lhs == null || inputs.rhs == null) {
+      return null;
+    }
+    const timestampDiff =
+      moment.utc(inputs.lhs).valueOf() - moment.utc(inputs.rhs).valueOf();
+    for (const unit of momentDurations) {
+      const duration = moment.duration(1, unit).asMilliseconds();
+      const diff = timestampDiff / duration;
+
+      if (diff >= 1) {
+        if (unit === 'years' || unit === 'months') {
+          return diff.toFixed(1) + ' ' + unit;
+        } else {
+          if (Math.floor(diff) === 1) {
+            return '1' + ' ' + unit.slice(0, -1);
```

Comment:
Why do we have this special case for "1" only?

Reply:
its so the unit shows for example: `1 day` instead of `1 days`

Reply:
👍